src/js/oTable.js

```diff
@@ -131,219 +23,347 @@ function descendingSort(...args) {
 	return 0 - ascendingSort.apply(this, args);
 }
 
-/**
- * Sorts the table by a specific column
- * @param  {number} The index of the column to sort the table by
- * @param  {bool} Which direction to sort in, ascending or descending
- * @param  {bool} Whether the values in this column are numeric, if they are numeric we convert the contents into numbers
- * @returns undefined
- */
-OTable.prototype.sortRowsByColumn = function (index, sortAscending, isNumericValue) {
-	const tbody = this.rootEl.querySelector('tbody');
-	const rows = Array.from(tbody.querySelectorAll('tr'));
-	const intlCollator = getIntlCollator();
-	rows.sort(function (a, b) {
-		let aCol = a.children[index];
-		let bCol = b.children[index];
-
-		if (aCol.getAttribute('data-o-table-order') !== null) {
-			aCol = aCol.getAttribute('data-o-table-order');
-			bCol = bCol.getAttribute('data-o-table-order');
-			if (!isNaN(parseInt(aCol, 10))) {
-				aCol = parseInt(aCol, 10);
-				bCol = parseInt(bCol, 10);
-			}
+function wrapElement(targetEl, wrapEl) {
+	const parentEl = targetEl.parentNode;
+	parentEl.insertBefore(wrapEl, targetEl);
+	wrapEl.appendChild(targetEl);
+}
+
+class OTable {
+	/**
+	 * Initialises an o-table components inside the element passed as the first parameter
+	 *
+	 * @param {(HTMLElement|string)} [el=document.body] - Element where to search for the o-table component. You can pass an HTMLElement or a selector string
+	 * @returns {OTable} - A single OTable instance
+	 */
+	constructor(rootEl) {
+		if (!rootEl) {
+			rootEl = document.body;
+		} else if (!(rootEl instanceof HTMLElement)) {
+			rootEl = document.querySelector(rootEl);
+		}
+		if (rootEl.getAttribute('data-o-component') === "o-table") {
+			this.rootEl = rootEl;
 		} else {
-			aCol = aCol.textContent;
-			bCol = bCol.textContent;
+			this.rootEl = rootEl.querySelector('[data-o-component~="o-table"]');
 		}
 
-		if (isNumericValue) {
-			aCol = parseFloat(aCol.replace(/,/g, ''));
-			bCol = parseFloat(bCol.replace(/,/g, ''));
-		}
+		if (this.rootEl !== undefined) {
+			this.listeners = [];
+			this.isResponsive = false;
+			this.rootEl.setAttribute('data-o-table--js', '');
+
+			// Map "data-o-table-order" to "data-o-table-sort-value".
+			const cellsWithOrder = this.rootEl.querySelectorAll('td[data-o-table-order], th[data-o-table-order]');
+			if (cellsWithOrder.length > 0) {
+				console.warn('o-table: "data-o-table-order" is deprecated, to provide a custom sort value for a table cell use "data-o-table-sort-value" instead.');
+				cellsWithOrder.forEach(cell => {
+					if (cell.getAttribute('data-o-table-order') !== null) {
+						cell.setAttribute('data-o-table-sort-value', cell.getAttribute('data-o-table-order'));
+					}
+				});
+			}
 
-		if (sortAscending) {
-			return ascendingSort(aCol, bCol, isNumericValue, intlCollator);
-		} else {
-			return descendingSort(aCol, bCol, isNumericValue, intlCollator);
+			this.tableHeaders = Array.from(this.rootEl.querySelectorAll('thead th'));
```

Comment:
I _think_ this will also match `th` elements which are not within a `thead` due to a quirk in querySelectorAll where by default it only verifies that the last element in the selector is within the search scope.

If we want to ensure we only return `th` elements which are within a `thead` we need to use `this.rootEl.querySelectorAll(':scope thead th')` however, that is not supported in IE or Edge and requires a polyfill being added to polyfill.io 😩

Reply:
I shall: 

```
const tHead = this.rootEl.querySelectorAll('thead')
this.tableHeaders = tHead.querySelectorAll('th')
```